models/room.go

```diff
@@ -394,26 +380,81 @@ func getReadyRooms(tx redis.Pipeliner, schedulerName string, size int, mr *Mixed
 	return res, err
 }
 
+func filterReadyAndOccupiedRooms(tx redis.Pipeliner, mr *MixedMetricsReporter, availableRooms *[]string, rooms []string, schedulerName string) error {
```

Comment:
Why receive `availableRooms`? I would rather build the slice and return it in this function.